### PR TITLE
[RFC] Cleanup & Improve Algolia indexing

### PR DESCRIPTION
**Pull request by mschinis**
**Solves #1**

**tldr**
This pull request does a number of improvements on the way the indexing is handled.

- [x] Improve code readability & code documentation
- [x] Clears the Algolia index before indexing, to avoid doing it manually
- [x] Use promises instead of callbacks when adding objects to Algolia

**For the future;**

The ideal solution to avoid search downtime, would be to avoid clearing the index before reindexing. A "_tmp" index can be created during the indexing operation, and once the indexing is complete it can be renamed.

The reason we cannot do this yet, is that we are currently just over 50% of our records quota, so creating a new index would mean we will go over our limit, causing the indexing operation to stop.

### CODE REVIEW

**Comment by sivakumar-kailasam** [comment]
Great to have you back in action again :)
